providers/google.go

```diff
@@ -222,3 +265,22 @@ func (g *GoogleOp) VerifyProvider(ctx context.Context, idt []byte, cic *clientin
 func (g *GoogleOp) HookHTTPSession(h http.HandlerFunc) {
 	g.httpSessionHook = h
 }
+
+// FindAvaliablePort attempts to open a listener on localhost until it finds one or runs out of redirectURIs to try
+func FindAvaliablePort(redirectURIs []string) (*url.URL, net.Listener, error) {
+	var ln net.Listener
+	var lnErr error
+	for _, v := range redirectURIs {
+		redirectURI, err := url.Parse(v)
+		if err != nil {
+			return nil, nil, fmt.Errorf("malformed redirectURI specified, redirectURI was %s", v)
+		}
+
+		lnStr := fmt.Sprintf("localhost:%s", redirectURI.Port())
+		ln, lnErr = net.Listen("tcp", lnStr)
+		if lnErr == nil {
+			return redirectURI, ln, nil
+		}
+	}
+	return nil, nil, fmt.Errorf("failed to start a listener for the callback from Google, got %w", lnErr)
```

Comment:
nit: I don't think you need to include `Google` in the error string; this logic is useful for other OPs too and I don't think the logic here is specific to Google

Reply:
Good point, fixed!